doc/api/customization.md

```diff
@@ -0,0 +1,60 @@
+# Customization
+
+> Stability: 1.0 - Early development
+
+Node.js has multiple APIs for customizing behavior and can easily be invoked via
+[`--import`][]. The package.json of such a plugin would look something like
+this:
+
+```json
+{
+  "name": "example-nodejs-plugin",
+  "keywords": [
+    "nodejs-plugin"
+  ],
+  "exports": {
+    "register": "./registration.mjs"
+  }
+}
+```
+
+Setting the keyword `nodejs-plugin` is important for users to find the
+package (which may be automated). It should also contain the official name(s)
+for the support it provides; a plugin extending Node.js support for TypeScript
+should contain `typescript` in its keywords; a customization package that does
+not extend Node.js support for TypeScript should not contain `typescript` (such
+as when the library is itself merely written in TypeScript). Incorrect keywords
+will cause a package to be listed for things it does not do, aggravating users.
```

Comment:
nit: I'm not fan of documenting non-technical patterns in the Node documentation.

Adding the npm keyword is useless from a technical perspective, whether users want to use them or not is more a subjective guideline than a "you must do this for the loader to be compatible with Node.js".

Reply:
The plan is that it would be a requirement for the wizard we plan to build, so that the wizard can find packages to suggest. But to your point, perhaps it’s better to exclude this until the wizard actually exists, and any steps that are required specifically for wizard compatibility (as opposed to general Node compatibility) could be added later as part of the PR that adds the wizard.

Reply:
> I'm not fan of documenting non-technical patterns in the Node documentation.

@arcanis are you talking about this part?

> Incorrect keywords will cause a package to be listed for things it does not do, aggravating users.

I'm trying to avoid users complaining (to node) of erroneous results from authors throwing in red-herrings.

---

What is described here is critical for the automated process (in whatever form). To limit changes for authors, these instructions need to be available as soon as possible.

> Adding the npm keyword is useless from a technical perspective, whether users want to use them or not is more a subjective guideline than a "you must do this for the loader to be compatible with Node.js".

I'm confused—this seems to directly contradict what you told me:

> I think uses the npm tags as part of the search, so if all loaders have similar tags that will probably help

So unless what Geoffrey said above (that tags are just keywords lifted from package.json) is incorrect, this is necessary.

The tag needs to be "claimed" immediately, regardless of how it gets there.